File: ragflow-automation/src/main/java/org/zemoso/base/ConfigReader.java
package org.zemoso.base;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

import static org.zemoso.utils.ConstantUtils.LOG_FAILED_TO_LOAD_PROPS;
import static org.zemoso.utils.ConstantUtils.PATH_TO_CONFIG;

public class ConfigReader {

    private static final Properties properties;

    static {
        try {
            FileInputStream fileInputStream = new FileInputStream(PATH_TO_CONFIG);
            properties = new Properties();
            properties.load(fileInputStream);
            fileInputStream.close();
        } catch (IOException e) {
            throw new RuntimeException(LOG_FAILED_TO_LOAD_PROPS, e);
        }
    }

    public static String getProperty(String key) {
        return properties.getProperty(key);
    }

    public static boolean getBooleanProperty(String key) {
        return Boolean.parseBoolean(getProperty(key));
    }
}

File: ragflow-automation-rearchitecture/selenium-csharp/Base/ConfigReader.cs
using AutomationC_.Utils;

namespace AutomationC_.Base;

public static class ConfigReader
{
    private static readonly Dictionary<string, string> Properties = LoadProperties();

    public static string GetProperty(string key) =>
        Properties.TryGetValue(key, out var value)
            ? value
            : throw new KeyNotFoundException($"Key '{key}' not found in application.properties");

    public static bool GetBooleanProperty(string key) =>
        bool.TryParse(GetProperty(key), out var result) && result;

    private static Dictionary<string, string> LoadProperties()
    {
        var properties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);

        var baseDirectory = AppContext.BaseDirectory;
        var configPath = Path.Combine(baseDirectory, ConstantUtils.PathToConfig.Replace("/", Path.DirectorySeparatorChar.ToString()));

        if (!File.Exists(configPath))
        {
            throw new FileNotFoundException(ConstantUtils.LogFailedToLoadProps, configPath);
        }

        foreach (var line in File.ReadAllLines(configPath))
        {
            var trimmed = line.Trim();

            if (string.IsNullOrWhiteSpace(trimmed) || trimmed.StartsWith("#"))
            {
                continue;
            }

            var separatorIndex = trimmed.IndexOf('=', StringComparison.Ordinal);
            if (separatorIndex <= 0)
            {
                continue;
            }

            var key = trimmed[..separatorIndex].Trim();
            var value = trimmed[(separatorIndex + 1)..].Trim();

            if (!string.IsNullOrEmpty(key))
            {
                properties[key] = value;
            }
        }

        return properties;
    }
}

